### DISCUSSION TITLE
Highlight/indicate which PRs will automerge

### DISCUSSION THREAD

**Discussion by MPV** [renovatebot/renovate#33168] [Suggest an Idea]
### Tell us more.

Could renovate show (without opening the PR) which PRs will be automerged?

For example in the:
- Onboarding PR (you will get these PRs AND they will automerge)
- Dependency Dashboard (when these PRs are created, they will be automerged)

For example, when onboarding many repositories using sharable presets that include some package rules which will trigger automerge, when developers get the onboarding PR it would be helpful if we/Renovate could help them understand which dependencies will be quickly rolled out (automerged) and which ones will need manual handling/review.

I've searched for similar requests, and the closest I've found is this one (which doesn't yet nail my specific need for it showing in the onboarding PR nor dependency dashboard, which labels wouldn't solve):
- https://github.com/renovatebot/renovate/discussions/24662

**Comment by MPV** [comment]
Alternatives considered:
1. "Manual detection"
   While it could be manually added to each `packageRule` that contains `automerge: true` it would be tedious to maintain across repos/presets/developers.
1. "Indicate with labels"
   As they only show on the PRs (not before their creation, i.e. in onboarding PR or dependency dashboard) it's not as easy to understand that triggering this kind of PR will also be automatically merged.
1. "Indicate with commit [prefix](https://docs.renovatebot.com/configuration-options/#commitmessageprefix) or [suffix](https://docs.renovatebot.com/configuration-options/#commitmessagesuffix)"
   As those are already used for other things (prefix for semantic commits, and suffix for other things) they might clash/overwrite other behaviour.

**Comment by MPV** [comment]
Maybe (as I guess [rarkins may have elluded to in #24662](https://github.com/renovatebot/renovate/discussions/24662#discussioncomment-7108272)?) we could have a [templating field](https://docs.renovatebot.com/templates/#other-available-fields) such as `isAutomerging` to allow this to be included in commit messages?

**Comment by MPV** [comment]
The expandable section for each dependency under "What to expect" in the onboarding PR is another place where it might be helpful to show whether a upcoming PR will be automerged (if not in the commit/PR title), i.e:
><img width="441" alt="Skärmavbild 2024-12-18 kl  13 41 29" src="https://github.com/user-attachments/assets/0ad5f132-0137-4aa2-95cd-bc95e8cc686b" />

**Comment by rarkins** [comment]
I think this work is dependent on "Dependency Dashboard Templating", similar to how we allowed for prBody templating long ago. We can then progressively expose various parts of the dashboard strings for templating. _Potentially_ changing defaults later too, if we like the ideas
